src/app/labeling/qgsmaptoollabel.cpp

```diff
@@ -1040,7 +1188,7 @@ bool QgsMapToolLabel::createAuxiliaryFields( QgsCalloutPosition &details, QgsCal
     {
       index = vlayer->fields().lookupField( prop.field() );
     }
-    else
+    else if ( prop.propertyType() != QgsProperty::ExpressionBasedProperty )
```

Comment:
```suggestion
    else
```
see above for reason